File: Part 1/rust_solution/src/trade.rs
use std::fmt;

#[derive(Clone, Debug, PartialEq)]
pub struct Trade {
    pub buyer: String,
    pub seller: String,
    pub instrument: String,
    pub quantity: i64,
    pub price: f32,
    pub bgen: usize,
    pub sgen: usize,
}

impl fmt::Display for Trade {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(
            f,
            "{}:{}:{}:{}:{}",
            self.buyer, self.seller, self.instrument, self.quantity, self.price,
        )
    }
}

File: Part 1/go_solution/trade.go
package main

import "fmt"

type Trade struct {
	Buyer      string
	Seller     string
	Instrument string
	Quantity   Quantity
	Price      Price
}

func (t Trade) String() string {
	return fmt.Sprintf("%v:%v:%v:%v:%v",
		t.Buyer, t.Seller, t.Instrument,
		t.Quantity, t.Price)
}
